engine/access/rpc/backend/backend_accounts.go

```diff
@@ -62,43 +69,57 @@ func (b *backendAccounts) GetAccountAtBlockHeight(
 		return nil, rpc.ConvertStorageError(err)
 	}
 
-	// get block ID of the header at the given height
+	// get the block id of the latest sealed header
 	blockID := header.ID()
 
-	account, err := b.getAccountAtBlockID(ctx, address, blockID)
+	account, err := b.getAccountAtBlock(ctx, address, blockID, header.Height)
 	if err != nil {
+		b.log.Debug().Err(err).Msgf("failed to get account at blockID: %v", blockID)
 		return nil, err
 	}
 
 	return account, nil
 }
 
-func (b *backendAccounts) getAccountAtBlockID(
+// getAccountAtBlock returns the account details at the given block
+//
+// The data may be sourced from the local storage or from an execution node depending on the nodes's
+// configuration and the availability of the data.
+func (b *backendAccounts) getAccountAtBlock(
 	ctx context.Context,
 	address flow.Address,
 	blockID flow.Identifier,
+	height uint64,
 ) (*flow.Account, error) {
-
-	exeReq := &execproto.GetAccountAtBlockIDRequest{
-		Address: address.Bytes(),
-		BlockId: blockID[:],
+	if b.scriptExecMode == ScriptExecutionModeExecutionNodesOnly {
+		return b.getAccountFromAnyExeNode(ctx, address, blockID)
 	}
 
-	execNodes, err := executionNodesForBlockID(ctx, blockID, b.executionReceipts, b.state, b.log)
-	if err != nil {
-		return nil, rpc.ConvertError(err, "failed to get account from the execution node", codes.Internal)
-	}
+	account, err := b.getAccountFromLocalStorage(ctx, address, height)
 
-	var exeRes *execproto.GetAccountAtBlockIDResponse
-	exeRes, err = b.getAccountFromAnyExeNode(ctx, execNodes, exeReq)
-	if err != nil {
-		b.log.Error().Err(err).Msg("failed to get account from execution nodes")
-		return nil, rpc.ConvertError(err, "failed to get account from the execution node", codes.Internal)
+	if err != nil && b.scriptExecMode == ScriptExecutionModeFailover {
```

Comment:
should we add a log here in case we get a successful response from the failover operation but not from the local operation?